lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -106,1360 +56,800 @@ function Install-Prerequisites {
-        Set-CharmState -Namespace "novahyperv" -Key "adapter_info" -Value $adapterInfo
-    }
-
-    return Get-RealInterface $port
-}
-
-function Get-DataPort {
-    # try and set up bonding early. This will create
-    # a new Net-LbfoTeam and try to acquire an IP address
-    # via DHCP. This interface may receive os-data-network IP.
-    $bondName = New-BondInterface
-    $managementOS = Get-JujuCharmConfig -Scope "vmswitch-management"
-
-    $net_type = Get-NetType
-    if ($net_type -eq "ovs"){
-        Write-JujuInfo "Trying to fetch OVS data port"
-        $dataPort = Get-OVSDataPort
-        return @($dataPort[0], $true)
-    }
-
-    if ($bondName) {
-        $adapter = Get-NetAdapter -Name bondName
-        return @($adapter, $managementOS)
-    }
-
-    Write-JujuInfo "Trying to fetch data port from config"
-    $nic = Get-InterfaceFromConfig
-    if(!$nic) {
-        $nic = Get-FallbackNetadapter
-        $managementOS = $true
-    }
-    $nic = Get-RealInterface $nic[0]
-    return @($nic[0], $managementOS)
-}
-
-function Start-ConfigureVMSwitch {
-    $VMswitchName = Juju-GetVMSwitch
-    $vmswitch = Get-VMSwitch -SwitchType External -Name $VMswitchName -ErrorAction SilentlyContinue
-
-    if($vmswitch){
-        return $true
-    }
-
-    $dataPort, $managementOS = Get-DataPort
-    $VMswitches = Get-VMSwitch -SwitchType External -ErrorAction SilentlyContinue
-    if ($VMswitches -and $VMswitches.Count -gt 0){
-        foreach($i in $VMswitches){
-            if ($i.NetAdapterInterfaceDescription -eq $dataPort.InterfaceDescription) {
-                Rename-VMSwitch $i -NewName $VMswitchName
-                Set-VMSwitch -Name $VMswitchName -AllowManagementOS $managementOS
-                return $true
-            }
-        }
-    }
-
-    Write-JujuInfo "Adding new vmswitch: $VMswitchName"
-    New-VMSwitch -Name $VMswitchName -NetAdapterName $dataPort.Name -AllowManagementOS $managementOS
-    return $true
-}
-
-function Download-File {
-    Param(
-        [Parameter(Mandatory=$true)]
-        [string]$url
-    )
-    $URI = [System.Uri]$url
-    $msi = $URI.segments[-1]
-    $download_location = Join-Path "$env:TEMP" $msi
-
-    if ($URI.fragment){
-        $fragment = $URI.fragment.Trim("#").Split("=")
-        if($fragment[0] -eq "md5"){
-            $md5 = $fragment[1]
-        }
-    }
-
-    $fileExists = Test-Path $download_location
-    if ($fileExists){
-        if ($md5){
-            $fileHash = (Get-FileHash -Algorithm MD5 $download_location).Hash
-            if ($fileHash -eq $md5){
-                return $download_location
-            }
-        }else{
-            return $download_location
+function Start-InstallHook {
+    if (!(Get-IsNanoServer)) {
+        try {
+            Set-MpPreference -DisableRealtimeMonitoring $true
+        } catch {
+            # No need to error out the hook if this fails.
+            Write-JujuWarning "Failed to disable antivirus: $_"
```

Comment:
lets change this to:

```
Failed to disable realtime monitoring
```

We can even do a version check and skip this for versions < windows 10/server 2016 and for Nano server.